### ISSUE TITLE
Pushing to an empty repository doesn't work

### ISSUE THREAD

**Issue by wmhilton** [isomorphic-git/isomorphic-git#27]
Need to handle case where the remote has no HEAD and no branches.

**Comment by hsablonniere** [comment]
Hey @wmhilton ;-)

I wanted to upgrade the clever-tools but it seems like 0.37.1 and after cannot push to empty repos (at least in my case). I works fine in 0.37.0.

Any ideas?

**Comment by hsablonniere** [comment]
Our git server seems to answer this:

```js
{
  result: {
    ok: [ 'unpack' ],
    errors: [ 'refs/heads/master missing necessary objects' ]
  }
}
```

**Comment by hsablonniere** [comment]
OK I managed to dirty fix the pb by hard coding mergebase to `'0000000000000000000000000000000000000000'` in push command.

**Comment by hsablonniere** [comment]
I'm going to close this issue as "technically pushing to an empty repo works, it's just not usable for our own projects and it's described in #1132
